### ISSUE TITLE
with-ajax makes changed.bs.select event to be called on search, not on change

### ISSUE THREAD

**Issue by ppazos** [truckingsim/Ajax-Bootstrap-Select#186]
We have a select picker with ajax, and a changed.bs.select event registered to check when the select value changes. Without ajax, the event is triggered only when the new value is selected, but with ajax the event is triggered when the search result is displayed, not when a new value is selected.

It seems the AJAX complete is calling a refresh(true)

https://github.com/truckingsim/Ajax-Bootstrap-Select/blob/master/src/js/classes/AjaxBootstrapSelectRequest.js#L110

And the refresh(true) is triggering the change event

https://github.com/truckingsim/Ajax-Bootstrap-Select/blob/master/src/js/classes/AjaxBootstrapSelectList.js#L243-L245

I don't think that is the correct behavior, since the change should only be triggered when the user actually selects a new value, not when the options are updated.

**Comment by Martijn-Bos** [comment]
I've also encountered the behaviour stated in the main comment.

**Comment by engin-can** [comment]
Experiencing same here...

**Comment by engin-can** [comment]
@ppazos - Any idea how to detect deselection of options? I am able to detect option changes, but not  deselection with `changed.bs.select`. It doesn't get triggered when there are no results found.

**Comment by engin-can** [comment]
Please take a look at this: https://stackoverflow.com/questions/61586202/how-to-catch-no-results-event-in-order-to-disable-input-field-in-bootstrap-sel

**Comment by ppazos** [comment]
@fbengo deselection is not related with this issue, this is about the search triggering a change event, but I guess for what you want you need to look for it in the code as I did to find the source of the issue I reported.

**Comment by engin-can** [comment]
I actually think it is related. I noticed that `changed.bs.select` is triggered when results for search query were found, and when an result option is selected. It is not triggered when no results were found. This behavior doesn't make sense to me. I tried to enable a different input field when an option is selected and disable it when no option is selected. I can't find a way to do this, because of this weird behavior...

![screenshot](https://user-images.githubusercontent.com/2794116/81025842-edf23480-8e2c-11ea-9eec-dec5589b84dc.gif)

**Comment by ppazos** [comment]
As mentioned before, the bug is the event to be triggered when a search is done, that shouldn't happen. That is the only weird behavior. Then what you want to do is not related with the issue, could be a feature request though.

We actually face the issue you are having, what should be done is to remove the event call from the search code (link provided above).

**Comment by engin-can** [comment]
I don't completely agree. It is not always triggered after search. That's my problem with it. It is only triggered when results were found, not when no results were found.

**Comment by ppazos** [comment]
@engin-can I agree, it's change is always triggered on search when there are result. Either way, is something that shouldn't happen since the change event should be triggered when the user changes the select picker, not when the search is executed and got results.

**Comment by ppazos** [comment]
BTW Bootstrap-Select released a new version with AJAX support, I guess this AJAX plugin is not longer needed.
